server.js

```diff
@@ -1,13 +1,217 @@
+  }
+
+  for (let requiredParam of [ 'color_one', 'color_two', 'color_three', 'color_four', 'color_five' ]) {
+    if (!patch[requiredParam]) {
+      return response.status(422).send({ error: `Expected format: { color_one:<String>, color_two:<String>, color_three:<String>, color_four:<String>, color_five:<String>} Your missing a ${[requiredParam]} property`});
+    }
+  }
+
+  for (let i = 0; i < patchKeys.length; i++) {
+    if(patchKeys[i] !== 'color_one' && patchKeys[i] !== 'color_two' && patchKeys[i] !== 'color_three' && patchKeys[i] !== 'color_four' && patchKeys[i] !== 'color_five') {
+      return response.status(422).send({ error: `Expected format: { color_one:<String>, color_two:<String>, color_three:<String>, color_four:<String>, color_five:<String>}, ${patchKeys[i]} is invalid property`});
+    }
+  }
+
+  try {
+    const paletteToPatch = await database('palettes').where('id', id).select();
+    if(!paletteToPatch.length) {
+      return response.status(404).json({ error: `Could not locate palette: ${id}` })
+    }
+    const updatedPalette = await database('palettes').where('id', id).update(patch, '*');
+    return response.status(200).json(updatedPalette);
+  } catch (error) {
+    return response.status(500).json({ error });
+  }
+});
+
+app.patch('/api/v1/projects/:id', async (request, response) => {
+  const { id } = request.params;
+  const patch = request.body;
+  const patchKeys = Object.keys(patch);
+
+  if(!parseInt(id)) {
+    return response.status(422).json({ error: `Incorrect ID: ${id}, Required data type: <Number>`});
+  }
+
+  for (let requiredParam of [ 'name' ]) {
+    if (!patch[requiredParam]) {
+      return response.status(422).send({ error: `Expected format: { name:<String> }, Your missing a ${[requiredParam]} property`});
+    }
+  }
+
+  for (let i = 0; i < patchKeys.length; i++) {
+    if(patchKeys[i] !== 'name') {
+      return response.status(422).send({ error: `Expected format: { name:<String> }, ${patchKeys[i]} is invalid property`});
+    }
+  }
+
+  try {
+    const projectToPatch = await database('projects').where('id', id).select();
+    if(!projectToPatch.length) {
+      return response.status(404).json({ error: `Could not locate project: ${id}` })
+    }
+    const updatedProject = await database('projects').where('id', id).update(patch, '*');
+    return response.status(200).json(updatedProject);
+  } catch (error) {
+    return response.status(500).json({ error });
+  }
+});
+
+app.delete('/api/v1/projects/:id', async (request, response) => {
+  const { id } = request.params;
+
+  if(!parseInt(id)) {
+    return response.status(422).json({ error: `Incorrect ID: ${id}, Required data type: <Number>`});
+  }
+
+  try {
+    const projectToDelete = await database('projects').where('id', id).select();
+    if(!projectToDelete.length) {
+      return response.status(404).json({ error: `Could not locate project: ${id}` });
+    }
+    const item = await database('projects').where('id', id).del();
```

Comment:
In regards to the issue. Not sure at all why that delete query needs to be assigned to a variable in order for it to work correctly. Good question!